servicetalk-transport-api/src/main/java/io/servicetalk/transport/api/ConnectionObserver.java

```diff
@@ -44,10 +44,25 @@
      */
     void onFlush();
 
+    /**
+     * Callback when a TCP handshake completes.
+     */
+    default void onTcpHandshakeComplete() {
```

Comment:
I would recommend avoid using `tcp` at this level bcz we could use the same abstractions for other protocols (udp, ..). Netty uses `channelActive` terminology and we could use something similar here.

Reply:
Agreed, changed it to `onTransportHandshakeComplete()`. Debated with myself if this callback really necessary or not. Discussed with @tkountis and he convinced me that it's a lot easier to understand API and events sequencing of the `ConnectionObserver` if we have it. Thanks for discussion!